fontc_crater/src/ci.rs

```diff
@@ -246,3 +263,44 @@ fn format_elapsed_time<Tmz: TimeZone>(start: &DateTime<Tmz>, end: &DateTime<Tmz>
     write!(&mut out, "{secs}s").unwrap();
     out
 }
+
+fn precompile_rust_binaries(temp_dir: &Path) -> (PathBuf, PathBuf) {
+    fn copy_file_into_dir(file_path: &Path, dir_path: &Path) -> PathBuf {
+        let new_file_path = dir_path.join(file_path.file_name().unwrap());
+        std::fs::copy(file_path, &new_file_path).expect("failed to copy binary to tempdir");
+        new_file_path
+    }
+
+    let fontc = compile_crate_or_die_trying("fontc");
+    let normalizer = compile_crate_or_die_trying("otl-normalizer");
+
+    (
+        copy_file_into_dir(&fontc, temp_dir),
+        copy_file_into_dir(&normalizer, temp_dir),
+    )
+}
+
+// if we can't compile fontc / otl-normalizer there's nothing we can do?
+fn compile_crate_or_die_trying(name: &str) -> PathBuf {
+    let status = Command::new("cargo")
+        .args(["build", "-p", name, "--release"])
+        .status()
+        .expect("failed to run cargo build");
+    if !status.success() {
+        panic!("cargo build '{name}' failed");
+    }
+
+    find_binary_target(name)
+}
+
+fn find_binary_target(name: &str) -> PathBuf {
```

Comment:
"find" implies a level of searching but we're just predicting, perhaps drop find_?

For consistency with compile or die perhaps it shoudl be locate_binary_or_die or some such?

Reply:
Also ... should this just be directly in compile crate?

Reply:
I mean it could be, but that doesn't really seem better in any meaningful way?